Simbody/src/CableSpan.cpp

```diff
@@ -0,0 +1,3028 @@
+                axes,
+                data.pathError);
+
+            // We can only use the path error jacobian if the small
+            // perturbation did not trigger any liftoff or touchdown on any
+            // obstacles. If any CurveSegment's WrappingStatus has changed, we
+            // will not continue with the test. Since the perturbation is
+            // small, this is unlikely to happen often.
+            bool WrappingStatusChanged = false;
+            {
+                int ix = -1;
+                for (const CurveSegment& curve : cable.m_curveSegments) {
+                    WrappingStatusChanged |=
+                        prevWrappingStatus.at(++ix) !=
+                        curve.getDataInst(sCopy).wrappingStatus;
+                }
+            }
+            if (WrappingStatusChanged) {
+                os << "Wrapping status changed: Stopping test\n";
+                break;
+            }
+
+            const Real predictionError =
+                (predictedPathError - data.pathError).norm();
+
+            bool passedTest = std::abs(predictionError / perturbation) <= bound;
+            if (!passedTest) {
+                os << "FAILED perturbation test for correction = "
+                   << data.pathCorrection << "\n";
+                os << "    Got      : " << data.pathError << "\n";
+                os << "    Predicted: " << predictedPathError << "\n";
+                os << "    Err      : "
+                   << (predictedPathError - data.pathError) / perturbation
+                   << "\n";
+                os << "    Err norm : "
+                   << (predictedPathError - data.pathError).normInf() /
+                          perturbation
+                   << "\n";
+            } else {
+                os << "PASSED perturbation test for correction = "
+                   << data.pathCorrection << "\n";
+                os << " ( error = "
+                   << (predictedPathError - data.pathError).normInf() /
+                          perturbation
+                   << " )\n";
+            }
+            success &= passedTest;
+        }
+    }
+    return success;
+}
+
+//==============================================================================
+//                          CableSubsystem::Impl
+//==============================================================================
+
+const CableSpan::Impl& CableSubsystem::Impl::getCableImpl(
+    CableSpanIndex index) const
+{
+    return getCable(index).getImpl();
+}
+
+int CableSubsystem::Impl::realizeSubsystemTopologyImpl(State& state) const
+{
+    // Briefly allow writing into the Topology cache. After this the
+    // Topology cache is const.
+    Impl* mutableThis = const_cast<Impl*>(this);
+
+    CableSpanData::Instance dataInst{};
+    CacheEntryIndex indexDataInst = allocateCacheEntry(
+        state,
+        Stage::Instance,
+        Stage::Infinity,
+        new Value<CableSpanData::Instance>(dataInst));
```

Comment:
Can be initialized here as an rvalue, rather than creating the temporary and copying it?